lib/core/client.js

```diff
@@ -863,7 +870,18 @@ function onSocketSession (session) {
 function connect (client) {
   assert(!client[kSocket])
 
-  const { protocol, port, hostname } = client[kUrl]
+  let { protocol, port, hostname } = client[kUrl]
+
+  // Resolve ipv6
+  if (hostname.startsWith('[')) {
+    const idx = hostname.indexOf(']')
+    const ip = idx !== -1 ? hostname.substr(1, idx - 1) /* istanbul ignore next */ : hostname
+
+    /* istanbul ignore else */
+    if (net.isIP(ip)) {
+      hostname = ip
+    }
```

Comment:
is the else possible here? what happens then?

Reply:
Same thing. I think I will add an assert here as well. Else should not be possible.

Reply:
Done!